### ISSUE TITLE
[locatorName]Wait should reject

### ISSUE THREAD

**Issue by johnkchiu** [paypal/nemo-view#6]
Not sure if I like `[locatorName]Wait` returns a promise that resolves to true/false.  This is little strange because when you write test and an element doesn't exists, you'll have to check the result.  For example:

```
nemo.view.SomeElementWait( function (find) {
    if (!find) {
        return done('not found');
    }
});
```

If `[locatorName]Wait` just returns a void promise and rejects, it'll be easier...

```
nemo.view.SomeElementWait();
```

Then the rejected exception can fail the test.

**Comment by grawk** [comment]
Hey @johnkchiu I think there's an error in your sample code. Should it be:

```
nemo.view.SomeElementWait().then(function (find) {
    if (!find) {
        return done('not found');
    }
});
```

Do you want to submit a PR for this?

**Comment by nikulkarni** [comment]
I took care of this in https://github.com/paypal/nemo-view/pull/9
So essentially Wait will throw error when the method doesn't find element in specified/default timeout unless you catch and handle it.

`nemo.view.SomeElementWait();` //will now throw error if it doesn't find

or if you want to handle error, you would do something like,

```
nemo.view.SomeElementWait().then( function () {
     //do stuff 
}, function (err) {
   done(err);
});
```

**Comment by johnkchiu** [comment]
It looks like changes in #10 now.  I like the new change.  Thanks.

**Comment by grawk** [comment]
This is released in nemo-view@0.4. Please update the following package versions to take advantage:
nemo-drivex@0.4
nemo-locatex@0.4
nemo@0.4
